common/src/stack/command/stack/commands/__init__.py

```diff
@@ -1137,6 +1138,68 @@ def characters(self, s):
 		self.text += s
 
 
+def get_mysql_connection(user=None, password=None):
+	"""
+	Creates a connection to MySQL and returns it, or returns None if
+	it can't connect. The caller must call `close` on the connection.
+	"""
+
+	connection = None
+
+	try:
+		if user is None:
+			# Root connects as 'apache', everyone else as the user
+			# running the python command.
+			if os.geteuid() == 0:
+				user = 'apache'
+			else:
+				user = pwd.getpwuid(os.geteuid())[0]
+
+		if password is None:
+			# Try to read the apache user's password
+			password = ''
+			try:
+				with open('/etc/apache.my.cnf') as f:
+					for line in f:
+						if line.startswith('password'):
+							password = line.split('=')[1].strip()
+							break
+			except:
```

Comment:
Consider only catching the expected exception here. A bare exception handler might catch and hide some other errors unintentionally.

`contextlib.suppress` could also be used in place of `try: ... except something: pass`.

Reply:
I'll consider it .... ..... ..... nope, going to still just catch `except`. If anything wrong happens in God's all-encompassing Universe trying to read the password from the file, I want to try a blank password.

Reply:
You will anger the PEP8 gods and catch `ctrl+c` though. I'd hate to see the PEP8ers come after you.